pkg/provider/provider.go

```diff
@@ -15,6 +15,7 @@ import (
 )
 
 const (
+	DefaultTimeFormat       = "2006-01-02T15:04:05.999999Z"
```

Comment:
Defaults should comply with SAML specification and be milliseconds and not micro seconds

`2006-01-02T15:04:05.999Z`

Reply:
that's exactly why this PR was created, as this format was used before I left this format as default to be backwards compatible, the used format in ZITADEL will be changed to be compliant with the SAML specification